### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by dignifiedquire**
- [x] Cleanup

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -87,4 +88,93 @@ describe('pull-length-prefixed', () => {
...
+        }
+
+        input.push(new Buffer(a.join('')))
+      }
+    })
+
+    it('encode - slow in - fast out', (done) => {
+      pull(
+        pull.values(input),
+        delay(10),
```
Comment (diasdavid) [inline test/index.spec.js]: Does it needs this delay to pass?

File: test/index.spec.js [hunk 2]
```diff
@@ -87,4 +88,93 @@ describe('pull-length-prefixed', () => {
...
+          done()
+        })
+      )
+    })
+
+    it('decode - slow in - fast out', (done) => {
+      pull(
+        pull.values(input),
+        lp.encode(),
+        delay(10),
```
Comment (diasdavid) [inline test/index.spec.js]: same question

File: test/index.spec.js (hunk 1 above)
Comment (dignifiedquire) [inline test/index.spec.js]: No, this is to simulate the connection in between being slow.

Comment (diasdavid) [inline test/index.spec.js]: What if the connection is fast? Was there a race condition at some point?

Comment (dignifiedquire) [inline test/index.spec.js]: I was just trying to simulate different scenarios. If the connection is fast it is like in the other cases, where encode is directly piped into decode.

**Pull request merged**